src/axom/spin/tests/spin_bvh.cpp

```diff
@@ -1108,6 +1109,103 @@ void check_single_box3d()
   axom::setDefaultAllocator(current_allocator);
 }
 
+//------------------------------------------------------------------------------
+
+/*!
+ * \brief Tests the find algorithm of the BVH in 3D.
+ *
+ *  A uniform mesh is used for this test where the query points are generated
+ *  by taking the centroids of the constituent cells of the mesh. Since the
+ *  mesh is a uniform, cartesian mesh the find algorithm should return exactly
+ *  one candidate for each query point corresponding to the cell on the mesh
+ *  that generated the centroid. This property is checked by using the
+ *  spin::UniformGrid class.
+ *
+ *  In addition, the test shifts the points by an offset to ensure that points
+ *  outside the mesh return no candidate.
+ *
+ */
```

Comment:
Thanks for adding the 3D test @kanye-quest !
Could you please also add a 2D version of this test?

This might help us figure out what might be needed for a 2D version that can take z-coordinate pointers that are `null`